parsl/monitoring/app_monitor.py

```diff
@@ -5,59 +5,87 @@
-            d['psutil_process_time_user'] += child.cpu_times().user
-            d['psutil_process_time_system'] += child.cpu_times().system
-            d['psutil_process_memory_virtual'] += to_mb(child.memory_info().vms)
-            d['psutil_process_memory_resident'] += to_mb(child.memory_info().rss)
+            d = {"psutil_process_" + str(k): v for k, v in pm.as_dict().items() if k in simple}
+            d["task_run_id"] = run_id
+            d["task_id"] = task_id
+            children = pm.children(recursive=True)
+            d["psutil_cpu_count"] = psutil.cpu_count()
+            d['psutil_process_memory_virtual'] = to_mb(pm.memory_info().vms)
+            d['psutil_process_memory_resident'] = to_mb(pm.memory_info().rss)
+            d['psutil_process_time_user'] = pm.cpu_times().user
+            d['psutil_process_time_system'] = pm.cpu_times().system
+            d['psutil_process_children_count'] = len(children)
             try:
-                d['psutil_process_disk_write'] += to_mb(child.io_counters().write_bytes)
-                d['psutil_process_disk_read'] += to_mb(child.io_counters().read_bytes)
+                d['psutil_process_disk_write'] = to_mb(pm.io_counters().write_bytes)
+                d['psutil_process_disk_read'] = to_mb(pm.io_counters().read_bytes)
             except psutil._exceptions.AccessDenied:
-                pass
-        logger.info("test", extra=d)
-        time.sleep(5)
+                # occassionally pid temp files that hold this information are unvailable to be read so set to zero
+                d['psutil_process_disk_write'] = 0
+                d['psutil_process_disk_read'] = 0
+            for child in children:
+                for k, v in child.as_dict(attrs=summable_values).items():
+                    d['psutil_process_' + str(k)] += v
+                d['psutil_process_time_user'] += child.cpu_times().user
+                d['psutil_process_time_system'] += child.cpu_times().system
+                d['psutil_process_memory_virtual'] += to_mb(child.memory_info().vms)
+                d['psutil_process_memory_resident'] += to_mb(child.memory_info().rss)
+                try:
+                    d['psutil_process_disk_write'] += to_mb(child.io_counters().write_bytes)
+                    d['psutil_process_disk_read'] += to_mb(child.io_counters().read_bytes)
+                except psutil._exceptions.AccessDenied:
+                    # occassionally pid temp files that hold this information are unvailable to be read so add zero
+                    d['psutil_process_disk_write'] += 0
+                    d['psutil_process_disk_read'] += 0
+
+            to_be_rounded = ['psutil_process_memory_virtual', 'psutil_process_memory_resident', 'psutil_process_cpu_percent', 'psutil_process_memory_percent',
+                             'psutil_process_disk_write', 'psutil_process_disk_read']
+            for k in to_be_rounded:
+                d[k] = round(d[k], 2)
```

Comment:
i'm not a fan of rounding deep inside stuff. i'm not going to really argue, though.

Reply:
This is similar to the change to MB (also the rounding of the times in the DFK logs), it was done in an effort to clean up the presentation through Elasticsearch. I am happy to remove this rounding.

Reply:
I agree with @benclifford